src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -44,1115 +44,716 @@
-
-                    if (FAILED(result))
-                    {
-                        err() << "Failed to verify DirectInput device axis mode for device "
-                              << std::quoted(m_identification.name.toAnsiString()) << ": " << result << std::endl;
-
-                        m_device->Release();
-                        m_device = nullptr;
-
-                        return false;
-                    }
-
-                    // If the axis mode hasn't been set to absolute fail here and blacklist the device
-                    if (property.dwData != DIPROPAXISMODE_ABS)
-                    {
-                        if (m_identification.vendorId && m_identification.productId)
-                        {
-                            JoystickBlacklistEntry entry{};
-                            entry.vendorId  = m_identification.vendorId;
-                            entry.productId = m_identification.productId;
-
-                            joystickBlacklist.push_back(entry);
-                            joystickBlacklist.shrink_to_fit();
-                        }
-
-                        m_device->Release();
-                        m_device = nullptr;
-
-                        return false;
-                    }
-
-                    break;
+                    joystick.m_xInputPacketNumber = xinputState.dwPacketNumber;
+
+                    auto& state       = joystick.m_state;
+                    auto& gamepad     = xinputState.Gamepad;
+                    state.buttons[0]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_A);
+                    state.buttons[1]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_B);
+                    state.buttons[2]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_X);
+                    state.buttons[3]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_Y);
+                    state.buttons[4]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_UP);
+                    state.buttons[5]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_DOWN);
+                    state.buttons[6]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_LEFT);
+                    state.buttons[7]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_DPAD_RIGHT);
+                    state.buttons[8]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_START);
+                    state.buttons[9]  = !!(gamepad.wButtons & XINPUT_GAMEPAD_BACK);
+                    state.buttons[10] = !!(gamepad.wButtons & XINPUT_GAMEPAD_LEFT_SHOULDER);
+                    state.buttons[11] = !!(gamepad.wButtons & XINPUT_GAMEPAD_RIGHT_SHOULDER);
+                    state.buttons[12] = !!(gamepad.wButtons & XINPUT_GAMEPAD_LEFT_THUMB);
+                    state.buttons[13] = !!(gamepad.wButtons & XINPUT_GAMEPAD_RIGHT_THUMB);
+
+                    const SHORT deadzone = XINPUT_GAMEPAD_LEFT_THUMB_DEADZONE / 4;
+
+                    state.axes[Joystick::Axis::X] = (abs(gamepad.sThumbLX) < deadzone) ? 0.0f : gamepad.sThumbLX / 327.670f;
```

Comment:
```suggestion
                    state.axes[Joystick::Axis::X] = (std::abs(gamepad.sThumbLX) < deadzone) ? 0.0f : gamepad.sThumbLX / 327.670f;
```
This applies to other uses of `abs` as well